### DISCUSSION TITLE
Trigger with release event and specific tag

### DISCUSSION THREAD

**Discussion by Petrakeas** [community/community#26603] [Actions]
<p>I want my workflow to be triggered when a release is published and has a specific tag syntax. So, I wrote the following:</p>
<pre><code class="lang-auto">on:
  release:
    types: [published]
    tags: 
      - v[0-9]+.*'
</code></pre>
<p>However, the workflow is triggered when a release is published, regardless of its tag name.</p>
<p>Is it possible to configure the filtering the way I want?</p>

**Comment by Simran-B** [comment]
<p>The <code>release</code> event doesn’t support filtering by tag name according to the docs. Only <code>push</code> and <code>pull_request</code> do:<br>
<a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="noopener nofollow ugc"><code>on.&lt;push|pull_request&gt;.&lt;branches|tags&gt;</code></a></p>
<p>You could add an <code>if</code> conditional to your jobs to skip them if the tag isn’t right. There will still be a workflow run however. It’s also non-trivial to do this since the the tag name pattern is a regular expressions but there is no workflow expression function to evaluate one. It should be possible to add a job that all other jobs depend on (<code>needs</code>), use it to validate the tag and set the result as job output.</p>
<p>The tag name should be accessible via <code>${{ github.event.release.tag_name }}</code>. You’ll need some shell code or <a href="https://github.com/actions/github-script" rel="noopener nofollow ugc">github-script</a> to evaluate the regular expression against the tag name.</p>

**Comment by Petrakeas** [comment, marked as answer]
<p><a class="mention" href="/simran-b">@Simran-B</a> thanks for the reply. I was hoping that filtering is added at the trigger level so that the workflow doesn’t even start.</p>
<p>I ended up using the <code>if</code> conditional this way:</p>
<pre><code class="lang-auto"> if:  ${{ startsWith(github.event.release.tag_name, 'some name') }}
</code></pre>

**Comment by akhilrajmailbox** [comment]
Do we have this feature? This is required as we must trigger the pipeline only for a specific release name format.

I am adding a new step to send a notification by forcefully breaking the pipeline as it always triggers for any release name.


```code
      - name: Validate Release
        env:
          RELEASE_NAME: ${{ github.event.release.name }}
          REGEX_PATTERN: "^v[0-9]+\\.[0-9]+\\.[0-9]+$"
        shell: bash
        run: |-
          if [[ "${{ env.RELEASE_NAME }}" =~ ${{ env.REGEX_PATTERN }} ]]; then
            echo "Valid version format: ${{ env.RELEASE_NAME }}"
          else
            echo "Invalid version format: ${{ env.RELEASE_NAME }}"
            exit 1
          fi
```
